x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransformWarnings.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.transform.action;
+
+import org.elasticsearch.xpack.core.transform.transforms.SourceConfig;
+import org.elasticsearch.xpack.core.transform.transforms.SyncConfig;
+import org.elasticsearch.xpack.transform.transforms.Function;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+class TransformWarnings {
```

Comment:
nit:

It would be good to scope this, e.g. some documentation about what this does as code doc. 

`TransformWarnings` sounds very generic. `TransformConfigLinter` ?

Reply:
Done.